base/REPLCompletions.jl

```diff
@@ -194,13 +213,20 @@ method_type_of_arg(val) = typeof(val)
 function complete_methods(ex_org::Expr)
     args_ex = Any[]
     for ex in ex_org.args # First ex is the function name
-        val = get_value(ex, Main)
-        val == nothing && return UTF8String[]
-        push!(args_ex, val)
+        val, found = get_value(ex, Main)
+        if length(args_ex) == 0
+            if !found || (found && !isgeneric(val))
+                return UTF8String[]
+            else
+                push!(args_ex, val)
+            end
+        else
+            # If a the argument could not be determined then it inserts type Any
+            found ? push!(args_ex, method_type_of_arg(val)) : push!(args_ex, Any)
+        end
```

Comment:
rather than use `length(args_ex) == 0` to block off the function name section, why not just explicitly write this code for the function name and then the rest? i.e.:
```julia
fn = get_value(ex_orgs.args[1], Main)
(fn == nothing || !isgeneric(fn)) && return UTF8String[]
for ex in ex_org.args[2:end]
   # simplified loop here...
```
might make it a little more clear.